crates/wasmtime/src/runtime/types.rs

```diff
@@ -2457,6 +2466,34 @@ impl GlobalType {
     }
 }
 
+// Tag Types
+
+/// A descriptor for a tag in a WebAssembly module.
+///
+/// This type describes an instance of a tag in a WebAssembly
+/// module. Tags are local to an [`Instance`](crate::Instance).
+#[derive(Debug, Clone, Hash)]
+pub struct TagType {
+    ty: FuncType,
+}
+
+impl TagType {
+    /// Creates a new global descriptor of the specified type.
+    pub fn new(ty: FuncType) -> TagType {
+        TagType { ty }
+    }
```

Comment:
This should validate that the function type only has parameters and not results, right? I'll add an item to https://github.com/bytecodealliance/wasmtime/pull/10177 for this, but feel free to remove it if my understanding is incorrect.

Reply:
Only for exception tags. But currently we don't have an exceptions feature. When stack switching is enabled the results can be nonempty.

Reply:
Gotcha, thanks!